File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
import math
import sys
input = sys.stdin.readline



# M 이상 N 이하의 소수를 모두 출력하는 프로그램 작성 

def is_prime(x):
    if x < 2:
        return False
    
    for i in range(2, int(math.sqrt(x)) + 1):
        if x % i == 0:
            return False
    
    return True


if __name__ == "__main__":
    M, N = map(int, input().split())

    for num in range(M, N+1):
        if is_prime(num):
            print(num)

File: 백준/Silver/1929. 소수 구하기/소수 구하기.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>
#include <limits.h>
#include <ctype.h>

#define TRUE 1
#define FALSE 0

int is_prime(int n) {
    if (n < 2) return FALSE; 
    if (n == 2) return TRUE; 
    if (n % 2 == 0) return FALSE; 

    for (int i = 3; i <= sqrt(n); i += 2) {
        if (n % i == 0) return FALSE;
    }
    return TRUE; 
}

int main() {
    int M, N;
    scanf("%d %d", &M, &N);

    for (int i = M; i <= N; i++) {
        if (is_prime(i)) {
            printf("%d\n", i);
        }
    }

    return 0;
}